src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -303,6 +305,84 @@ CreateDistributedInsertSelectPlan(Query *originalQuery,
 	return distributedPlan;
 }
 
+/*
+ * CreateInsertSelectIntoLocalTablePlan
+ */
+DistributedPlan *
+CreateInsertSelectIntoLocalTablePlan(uint64 planId, Query *originalQuery, PlannerRestrictionContext *plannerRestrictionContext)
+{
+	RangeTblEntry *selectRte = ExtractSelectRangeTableEntry(originalQuery);
+	//RangeTblEntry *insertRte = ExtractResultRelationRTE(originalQuery);
+	//Oid targetRelationId = insertRte->relid;
+	Query * selectQuery = selectRte->subquery;
+	DistributedPlan * distPlan = NULL;
+
+	distPlan = CreateRouterPlan(selectQuery, selectQuery, plannerRestrictionContext);
```

Comment:
Can you call `CreateDistributedPlan` here and set the masterQuery afterwards if it is NULL? That way, we can exercise all our existing planning logic.